src/main/string_util.java

```diff
@@ -1,23 +1,13 @@
 package main;
 
+import javax.xml.bind.DatatypeConverter;
+
 public class string_util {
-    public static byte[] toByteArray(String s) {
-        int len = s.length();
-        byte[] data = new byte[len / 2];
-        for (int i = 0; i < len; i += 2) {
-            data[i / 2] = (byte) ((Character.digit(s.charAt(i), 16) << 4)
-                                 + Character.digit(s.charAt(i+1), 16));
-        }
-        return data;
+    public static byte[] hexStringToByteArray(String s) {
+          return DatatypeConverter.parseHexBinary(s);

```

Comment:
i also used this and i notice that it doesn't produce the right conversion with some key

Reply:
LOL, that's amazing, since I had the same issue with another one, will see if I can fix the explicit version for my key instead.

Reply:
it's ethier broken the extraction in the first place or the conversion LOL...

Reply:
news?

Reply:
I'm some kilometers away from the java sdk atm, will work on it _soon_
Meanwhile, could you point me to that some problematic key you mentioned?